diff --git a/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt b/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
index be26a22..a2834c3 100644
--- a/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
+++ b/node_modules/@react-native/gradle-plugin/settings-plugin/src/main/kotlin/com/facebook/react/ReactSettingsExtension.kt
@@ -149,7 +149,7 @@ abstract class ReactSettingsExtension @Inject constructor(val settings: Settings
       if (isCacheDirty(cacheJsonConfig, cacheFolder, lockFiles)) {
         val process = updateJsonConfig.start()
 
-        val finished = process.waitFor(5, TimeUnit.MINUTES)
+        val finished = process.waitFor(30, TimeUnit.MINUTES)
         if (!finished || (process.exitValue() != 0)) {
           val command = updateJsonConfig.command().joinToString(" ")
           val prefixCommand = "ERROR: autolinkLibrariesFromCommand: process $command"
